### PR TITLE
GuidedSearcher and GuidedMode

### PR DESCRIPTION
**Pull request by misonijnik**

### CODE REVIEW

File: VSharp.SILI/CILState.fs [hunk 1]
```diff
@@ -98,6 +114,12 @@ module internal CilStateOperations =
 
     let currentOffset = currentIp >> offsetOf
 
+    let inCoverageZone coverageZone startingLoc loc =
+        match coverageZone with
+        | MethodZone -> loc.method = startingLoc.method
+        | ClassZone -> loc.method.DeclaringType= startingLoc.method.DeclaringType
```
Comment (misonijnik) [inline VSharp.SILI/CILState.fs:120]: loc.method.DeclaringType = startingLoc.method.DeclaringType

File: VSharp.SILI/CFG.fs [hunk 2]
```diff
@@ -205,14 +247,82 @@ module public CFG =
...
+        else false
+
+    let hasSiblings (method : MethodBase) (offset : offset) =
+        if not method.IsAbstract then
+            let cfg = findCfg method
+            if cfg.sortedOffsets.BinarySearch(offset) >= 0 then
+                let parents = cfg.reverseGraph.[offset]
+                parents |> Seq.exists (fun parent ->
+                let count = cfg.graph.[parent].Count
+                ()
```
Comment (misonijnik) [inline VSharp.SILI/CFG.fs:322]: clean

File: VSharp.SILI/TargetedSearcher.fs [hunk 3]
```diff
@@ -2,333 +2,186 @@ namespace VSharp.Interpreter.IL
...
+                | Some loc -> CFG.isVertex loc.method loc.offset
+                | None -> false
+
+            isStopped state  ||  onVertex state
+
+        if needsUpdating parent then
+            base.Update (parent, newStates)
+        else
+            base.Insert newStates
+        for state in Seq.append [parent] newStates do
```
Comment (misonijnik) [inline VSharp.SILI/TargetedSearcher.fs:37]: space

File: VSharp.Utils/PriorityCollection.fs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    abstract member Insert : 'a -> uint -> unit
+    abstract member Remove : 'a -> unit
+    abstract member Update : 'a -> uint -> unit
+    abstract member Choose : uint -> 'a option
+    abstract member Contains : 'a -> bool
+    abstract member TryGetPriority : 'a -> uint option
+    abstract member MaxPriority : uint
+    abstract member ToSeq : 'a seq
+
+type BidicitionaryPriorityQueue<'a when 'a : equality>() =
```
Comment (misonijnik) [inline VSharp.Utils/PriorityCollection.fs:18]: Bidictionary

**Review by dvvrd — APPROVED** [review]

**Pull request merged**